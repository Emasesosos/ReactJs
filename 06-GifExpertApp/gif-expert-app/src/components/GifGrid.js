import React from 'react';
import PropTypes from 'prop-types';
import useFetchGifs from '../hooks/useFetchGifs';
import Chase from './../spinners/Chase';
import GifGridItem from './GifGridItem';
import Imagen from './Imagen';

const GifGrid = ({category}) => { 

    const { data:images, loading } = useFetchGifs(category);

    return (
        <>
            <div className="category animate__animated animate__fadeIn">
                <Imagen />
                <h3>{ category }</h3>
                <Imagen />
            </div>
            { loading && <Chase /> }
            <div className="card-grid">
                {
                    images.map(img => {
                            return <GifGridItem 
                                        key={img.id}
                                        {...img}                       
                                    />;
                        }
                    )
                }
            </div>
        </>
    )
};

GifGrid.propTypes = {
    category: PropTypes.string.isRequired,
}

export default GifGrid;
