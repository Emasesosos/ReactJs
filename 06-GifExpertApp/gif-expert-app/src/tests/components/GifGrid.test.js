import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid from './../../components/GifGrid';
import useFetchGifs from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('Puebas en <GifGrid />', () => {

    const category = 'Goku';

    test('Debe de mostrar el componente correctamente', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true,
        });
        
        const wrapper = shallow( <GifGrid category={category}/> );
        expect(wrapper).toMatchSnapshot();

    });

    test('Debe de mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {

        const gifs = [
            {
                id: 'ABC',
                url: 'https://localhost/cualquiercosa/cosa.jpg',
                title: 'Cualquier cosa',
            },
            {
                id: '123',
                url: 'https://localhost/cualquiercosa/cosa.jpg',
                title: 'Cualquier cosa',
            }
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false,
        });
        
        const wrapper = shallow( <GifGrid category={category}/> );
        const ChaseComponent = wrapper.find('Chase').exists();
        const GifGridItemComponent = wrapper.find('GifGridItem');

        expect(wrapper).toMatchSnapshot();
        expect(ChaseComponent).toBe(false);
        expect(GifGridItemComponent.length).toBe(gifs.length); 

    });

});