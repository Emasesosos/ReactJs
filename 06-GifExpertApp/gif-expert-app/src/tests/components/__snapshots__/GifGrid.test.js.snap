// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Puebas en <GifGrid /> Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <div
    className="category animate__animated animate__fadeIn"
  >
    <Imagen />
    <h3>
      Goku
    </h3>
    <Imagen />
  </div>
  <Chase />
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Puebas en <GifGrid /> Debe de mostrar items cuando se cargan im√°genes useFetchGifs 1`] = `
<Fragment>
  <div
    className="category animate__animated animate__fadeIn"
  >
    <Imagen />
    <h3>
      Goku
    </h3>
    <Imagen />
  </div>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa/cosa.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Cualquier cosa"
      url="https://localhost/cualquiercosa/cosa.jpg"
    />
  </div>
</Fragment>
`;
